src/gui/LayoutMapper.cpp

```diff
@@ -146,48 +131,64 @@ int LayoutMapper::getPairedPages()
 int LayoutMapper::map(int x, int y)
 {
 	XOJ_CHECK_TYPE(LayoutMapper);
+	if (this->layoutType >= Size)
+	{
+		g_warning("LayoutMapper::Unknown layout: %d\n", this->layoutType);
+		return 0;
+	}
 
-	if (this->layoutType < BitFlagsUsedToHere)
+	int res;
+	if (isRightToLeft())
 	{
-		int res;
-		if (this->layoutType & LayoutBitFlags::RightToLeft)
-		{
-			// reverse x
-			x = this->cols - 1 - x;
-		}
+		// reverse x
+		x = this->cols - 1 - x;
+	}
 
-		if (this->layoutType & LayoutBitFlags::BottomToTop)
-		{
-			// reverse y
-			y = this->rows - 1 - y;
-		}
+	if (isBottomToTop())
+	{
+		// reverse y
+		y = this->rows - 1 - y;
+	}
 
-		if (this->layoutType & LayoutBitFlags::Vertically)
+	if (isVertical())
+	{
+		if (this->isPairedPages)
 		{
-			if (this->isPairedPages)
-			{
-				res = ((y + this->rows * ((int) (x / 2))) * 2) + x % 2;
-			}
-			else
-			{
-				res = y + this->rows * x;
-			}
+			res = ((y + this->rows * ((int) (x / 2))) * 2) + x % 2;
 		}
-		else //Horizontal
+		else
 		{
-			res = x + this->cols * y;
+			res = y + this->rows * x;
 		}
+	}
+	else //Horizontal
+	{
+		res = x + this->cols * y;
+	}
 
-		res -= this->offset;
-
-		if (res >= this->actualPages)
-		{
-			res = -1;
-		}
+	res -= this->offset;
 
-		return res;
+	if (res >= this->actualPages)
+	{
+		res = -1;
 	}
 
-	g_warning("LayoutMapper::Unknown layout: %d\n", this->layoutType);
-	return 0;
+	return res;
+
+
 }
+
+bool LayoutMapper::isVertical() const
+{
+	return (layoutType & Vertical) == Vertical;
+}
+
+bool LayoutMapper::isBottomToTop() const
+{
+	return (layoutType & BottomToTop) == BottomToTop;
+}
+
+bool LayoutMapper::isRightToLeft() const
+{
+	return (this->layoutType & RightToLeft) == RightToLeft;
```

Comment:
Nitpicking, but please be consistent with `this->layoutType` or `layoutType` here.

Reply:
Since we don't mark member variables with a "_" postfix it is better to use "this->variable". I don't think this is nitpicking ;) .